cmd/agent/app/config.go

```diff
@@ -75,9 +76,8 @@ type Builder struct {
 	Processors     []ProcessorConfiguration    `yaml:"processors"`
 	SamplingServer SamplingServerConfiguration `yaml:"samplingServer"`
 
-	// CollectorHostPort is the hostPort for Jaeger Collector.
-	// Set this to communicate with Collector directly, without routing layer.
-	CollectorHostPort string `yaml:"collectorHostPort"`
+	// CollectorHostPorts is the static list of hostPorts for Jaeger Collector.
+	CollectorHostPorts string `yaml:"CollectorHostPorts"`
```

Comment:
This will be a breaking change. Also, in YAML it would be more natural to declare this as `[]string`. So I would suggest the following
  1. keep `collectorHostPort` but document it as comma-separated string
  1. change  CollectorHostPorts to `[]string` (yaml label should start with lower-case)
  1. leave the builder binding the string flag to the original `collectorHostPort`
  1. in the constructor, combing values from `[]string` and from `collectorHostPort` (after splitting on `,`)